cpp/test/Ice/invoke/AllTests.cpp

```diff
@@ -434,14 +442,63 @@ allTests(Test::TestHelper* helper)
         test(completed.get_future().get());
     }
 
+    {
+        // Call oneway with a regular (twoway) proxy
+        promise<pair<bool, vector<Ice::Byte>>> completed;
+        Ice::ByteSeq inEncaps, outEncaps;
+        cl->ice_invokeAsync(
+            "opOneway",
+            Ice::OperationMode::Normal,
+            inEncaps,
+            [&](bool ok, vector<Ice::Byte> ouEncaps)
+            {
+                completed.set_value(make_pair(ok, std::move(ouEncaps)));
+            },
+            [&](exception_ptr ex)
+            {
+                completed.set_exception(ex);
+            });
+        pair<bool, vector<Ice::Byte>> result = completed.get_future().get();
+        test(result.first);
+        test(result.second.size() == 6); // Empty encapsulation
+    }
+
     //
     // repeat with the future API.
     //
 
     {
+        // Oneway operation with a twoway proxy must return empty encapsulation.
```

Comment:
This test worked fine before the fix, oneway operation invoked from twoway proxy returns an empty encapsulation written in 6 bytes.